<!-- -*- mode: XML -*- -->
<launch>
    <arg name="connect_on_startup"   default="true"/>
    <arg name="calibrate_on_startup" default="true"/>
    <arg name="engage_on_startup"    default="true"/>

    <arg name="publish_odom"        default="true"/>
    <arg name="publish_odom_tf"     default="false"/>
    
    <arg name="output"   default="screen"/>
    <arg name="respawn"   default="false"/>
    
    <node pkg="odrive_ros" type="odrive_node" name="odrive" ns="odrive" 
            output="$(arg output)" respawn="$(arg respawn)">
        <param name="wheel_track" value="0.285"/>
        <param name="tyre_circumference" value="0.341"/>
        
        <param name="connect_on_startup"   value="$(arg connect_on_startup)"/>
        <param name="calibrate_on_startup" value="$(arg calibrate_on_startup)"/>
        <param name="engage_on_startup"    value="$(arg engage_on_startup)"/>
        
        <param name="publish_odom"         value="$(arg publish_odom)"/>
        <param name="publish_odom_tf"      value="$(arg publish_odom_tf)"/>
    </node>
</launch>
